aquarius/events/processors.py

```diff
@@ -353,3 +353,38 @@ def process(self):
         self.es_instance.update(self.asset, self.did)
 
         return self.asset
+
+
+class MetadataStateProcessor(EventProcessor):
+    def process(self):
+        txid = self.txid
+
+        decryptor_url, *_ = self.dt_contract.caller.getMetaData()
+
+        asset = decrypt_ddo(
+            web3=self._web3,
+            provider_url=decryptor_url,
+            contract_address=self.event.address,
+            chain_id=self._chain_id,
+            txid=txid,
+        )
+
+        if not self.check_document_hash(asset):
+            return False
+
+        self.did = asset["id"]
+
+        if self.event.args.state == 0:
```

Comment:
please only decrypt the asset if we are not deleting it. Let's not make provider calls if not necessary. I would suggest handling the simpler case first, e.g.

if self.event.args.state != 0:
self._es_instance.delete(asset["id"])
return

then you don't need the else block, just its contents. (forcing return from the first if block means there is no other flow)

Reply:
Actually, thinking about it again, I am not even sure you need to decrypt it otherwise. If the MetadataCreatedProcessor args match properly, then that processor will do the decrpyption.